cobc/cobc.c

```diff
@@ -1316,6 +1316,61 @@ cobc_plex_strdup (const char *dupstr)
 	return p;
 }
 
+/* Return a newly allocated string with only the first len chars of the first argument */
+void *
+cobc_plex_strsub (const char *s, const int len)
+{
+	void	*p;
+	int	n;
+
+#ifdef	COB_TREE_DEBUG
+	/* LCOV_EXCL_START */
+	if (unlikely (!s)) {
+		cobc_err_msg (_("call to %s with NULL pointer"),
+			      "cobc_plex_strsub");
+		cobc_abort_terminate (1);
+	}
+	/* LCOV_EXCL_STOP */
+#endif
+	n = strlen (s);
+
+#ifdef	COB_TREE_DEBUG
+	/* LCOV_EXCL_START */
+	if ( len>n ) {
+		cobc_err_msg (_("call to %s with bad argument len=%d>%d=strlen(s)"),
```

Comment:
please ensure that these internal-only messages are normally _not_ gettextized, because a user should never see them and translators also tend to don't want to translate those (and if they are translated and we get bug reports, then we often don't understand it either and need to reverse-lookup the msgid); there are exceptions like the one above - but this is also used _very_ often and is an existing old msgid

Reply:
done